15-components-lab_v16/src/02-numberguessing.jsx

```diff
@@ -12,6 +12,47 @@
  */
 
 import React from 'react';
+import PropTypes from 'prop-types';
 
-export default React.createClass({
-});
+export default class NumberGuessing extends React.Component {
+
+    static propTypes = {
+        num: PropTypes.number,
+        rand: PropTypes.number
+      };
+    
+      static defaultProps = {
+        num: Math.floor((Math.random() * 1000) + 1),
+        rand: Math.floor((Math.random() * 5) + 1)
+      };
+
+    constructor(props){
+        super(props);
+        this.state = {message : '', guess : 0};
```

Comment:
I wouldn't keep the actual message text in state, but would rather use a key to a texts object of some sort. For example:

```
const MESSAGES = { HIGH: 'too high', LOW: 'too low', BLANK: '', WIN: 'You Win!' };
```

And then in the constructor:

```
this.state = { message: MESSAGES.BLANK };
```